Cyra/lab2/part3/flake.nix

```diff
@@ -0,0 +1,33 @@
+{
+  description = "flake part 3";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
+    utils.url = "github:numtide/flake-utils";
+    hello_lib.url = "github:skyraal/nixlab2";
+  };
+
+  outputs = { self, nixpkgs, utils, hello_lib }:
+  let
+    pkgs = import nixpkgs {
+      system = "aarch64-darwin";
+      overlays = [ hello_lib.overlays.default ];
+    };
+
+   in {
+    packages.aarch64-darwin.default = pkgs.hello_app;
+    devShells.aarch64-darwin.default = pkgs.mkShell rec {
+      name = "nix-devshell";
+      packages = with pkgs; [
+        cmake
+        hello_app
+      ];
```

Comment:
your flake input is good and you are using your overlay correctly that you define, however just put your executable derivation / code in this repo and build add it to this flake, dont just put everything in your upstream